File: app/src/main/java/ru/merkulyevsasha/news/data/db/NewsDbRepository.java
package ru.merkulyevsasha.news.data.db;

import java.util.List;

import io.reactivex.Single;
import ru.merkulyevsasha.news.pojos.Article;

public interface NewsDbRepository {

    void addListNews(List<Article> items);

    void delete(int navId);

    void deleteAll();

    Single<List<Article>> selectAll();

    Single<List<Article>> selectNavId(int navId);

    Single<List<Article>> search(String searchTtext);

}

File: app/src/main/java/ru/merkulyevsasha/news/data/NewsDbRepository.kt
package ru.merkulyevsasha.news.data

import java.util.Date

import io.reactivex.Single
import ru.merkulyevsasha.news.models.Article

interface NewsDbRepository {

    val lastPubDate: Date?

    fun addListNews(items: List<Article>)

    fun delete(navId: Int)

    fun deleteAll()

    fun selectAll(): Single<List<Article>>

    fun selectNavId(navId: Int): Single<List<Article>>

    fun search(searchTtext: String): Single<List<Article>>
}
